### PR TITLE
Fix restoring software license dialog

### PR DESCRIPTION
**Pull request by TobiGr**
#### What is it?
- [x] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
The last viewed license was re-opened after an orientation change in the About and License Fragments even if the license view/dialog has been closed by the user earlier. I needed to move the method creating the AlertDialog into the Fragment class to access `activeLicense`. An alternative implementation would have been to pass the listener as variable. But I think that the AlertDialog creation is part of the Fragment and should therefore be in the LicenseFragment and not a helper class.

The dialog buttons to open the software's website is also restored correctly after a rotation change now.

#### Before/After Screenshots/Screen Record

| Before | After |
|--------|--------|
| https://github.com/TeamNewPipe/NewPipe/assets/17365767/6efedaaf-8680-4481-be06-0ef86357ce21 | https://github.com/TeamNewPipe/NewPipe/assets/17365767/82f54007-f2c3-4139-8f87-a2452daa5eba | 

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR. You can find more info and a video demonstration [on this wiki page](https://github.com/TeamNewPipe/NewPipe/wiki/Download-APK-for-PR).

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Stypox — CHANGES_REQUESTED** [review]
The neutral button "open website license" disappears on rotation, right? Since you don't distinguish between `SoftwareComponent` and `License`. Also, setting the `activeLicense` should happen from inside `showLicense`, not right before calling it.

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragment.kt [hunk 1]
```diff
@@ -57,26 +66,73 @@ class LicenseFragment : Fragment() {
...
+                    val webView = WebView(context)
+                    webView.loadData(webViewData, "text/html; charset=UTF-8", "base64")
+
+                    Localization.assureCorrectAppLanguage(context)
+                    val builder = AlertDialog.Builder(requireContext())
+                        .setTitle(softwareComponent.name)
+                        .setView(webView)
+                        .setOnCancelListener { activeSoftwareComponent = null }
+                        .setOnDismissListener { activeSoftwareComponent = null }
+                    if (softwareComponent == NEWPIPE_SOFTWARE_COMPONENT) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/about/LicenseFragment.kt:109]: I think it's perfectly fine if you show "dismiss" and "open website" even for NewPipe, to further simplify the code and make things consistent. After solving (or dismissing) this comment feel free to merge directly, thanks! The code looks much better

**Review by Stypox — APPROVED** [review]

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragment.kt (hunk 1 above)
Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/about/LicenseFragment.kt:109]: I think it is ok to handle NewPipe differently and not show the website button. We already do that in the AboutFragment.
However, we can use the same text on the positive button to close the dialog. I'd not use "ok", but rather "dismiss" or even better just "close".
 @opusforlife2 What button text would you use to close the license dialog? _ok_, _dismiss_ or _close_

Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/about/LicenseFragment.kt:109]: "done" was not an option 😂

Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/about/LicenseFragment.kt:109]: Done is also ok

**Pull request merged**